fv3core/fv3core/_config.py

```diff
@@ -264,6 +269,99 @@ class DynamicalCoreConfig:
     nf_omega: int = NamelistDefaults.nf_omega
     fv_sg_adj: int = NamelistDefaults.fv_sg_adj
     n_sponge: int = NamelistDefaults.n_sponge
+    namelist_override: str = None
+
+    def __post_init__(self):
+        if self.namelist_override is not None and os.path.exists(
+            self.namelist_override
+        ):
+            f90_nml = f90nml.read(self.namelist_override)
+            namelist = Namelist.from_f90nml(f90_nml)
+            self.dt_atmos = namelist.dt_atmos
+            self.a_imp = namelist.a_imp
+            self.beta = namelist.beta
+            self.consv_te = namelist.consv_te
+            self.d2_bg = namelist.d2_bg
+            self.d2_bg_k1 = namelist.d2_bg_k1
+            self.d2_bg_k2 = namelist.d2_bg_k2
+            self.d4_bg = namelist.d4_bg
```

Comment:
Is there a way to have post_init return a new class and override the initialization? I tried a few variations of calling class methods but no luck, so ended up settling on explicitly setting each variable.

Reply:
I expected you'd set each variable here. If you like you can

1) make a new object in here with DynamicalCoreConfig.from_f90nml
2) iterate over the dataclass fields of the object or self (they're the same type) and assign with setattr and getattr, this is OK type-wise because you're assigning from and to the same type without adding new attributes.

Reply:
Great suggestion, I knew there was a better way to utilize `from_f90nml`! Updating this.